src/lib/Select.svelte

```diff
@@ -1,90 +1,316 @@
+                    itemFilter,
+                    convertStringItemsToObjects,
+                    filterGroupedItems,
+                })),
+        );
+    });
+    $effect.pre(() => {
+        prev_multiple = multiple;
+    });
+    $effect(() => {
+        hasValue;
+        untrack(() => {
+            if (items) setValue(items);
+        });
+    });
+    $effect(() => {
+        searchable;
+        untrack(() => {
+            if (inputAttributes || !searchable) assignInputAttributes();
+        });
+    });
+    $effect(() => {
+        untrack(() => {
+            if (multiple) setupMulti();
+        });
+    });
+    $effect(() => {
+        if (prev_multiple && !multiple) setupSingle();
+    });
+    $effect(() => {
+        if (multiple && value && value.length > 1) checkValueForDuplicates();
+    });
+    $effect(() => {
+        if (value) dispatchSelectedItem();
+    });
+    $effect(() => {
+        if (prev_value && !value) {
+            oninput(value || []);
+        }
+    });
+    $effect(() => {
+        if (!focused && input) closeList();
+    });
+    $effect(() => {
+        filterText;
+        untrack(() => {
+            if (filterText !== prev_filterText) setupFilterText();
+        });
+    });
+    $effect(() => {
+        if (!multiple && listOpen && value && filteredItems) setValueIndexAsHoverIndex();
+    });
+    $effect(() => {
+        dispatchHover(hoverItemIndex);
+    });
+    $effect(() => {
+        multiple;
+        value;
+        untrack(() => (hasValue = multiple ? value && value?.length > 0 : !!value));
+    });
+    $effect(() => {
+        hideSelectedItem = hasValue && filterText.length > 0;
+    });
+    $effect(() => {
+        showClear = hasValue && clearable && !disabled && !loading;
+    });
+    $effect(() => {
+        ariaSelection = value ? handleAriaSelection(multiple) : '';
+    });
+    $effect(() => {
+        ariaContext = handleAriaContent({ filteredItems, hoverItemIndex, focused, listOpen });
+    });
+    $effect(() => {
+        multiple;
+        itemId;
+        value;
+        untrack(() => {
+            justValue = computeJustValue();
+        });
+    });
 
+    $effect(() => {
+        filteredItems;
+        value;
+        multiple;
+        untrack(() => {
+            if (listOpen && filteredItems && !multiple && !value) checkHoverSelectable();
+        });
+    });
+    $effect(() => {
+        if (filteredItems) handleFilterEvent(filteredItems);
+    });
+    $effect(() => {
+        if (container && floatingConfig) floatingUpdate(Object.assign(_floatingConfig, floatingConfig));
+    });
+    $effect(() => {
+        list;
+        untrack(() => (listDom = !!list));
+    });
+    $effect(() => {
+        listOpen;
+        untrack(() => listMounted(list, listOpen));
+    });
+    $effect(() => {
+        if (listOpen && container && list) setListWidth();
+    });
+    $effect(() => {
+        scrollToHoverItem = hoverItemIndex;
```

Comment:
`scrollToHoverItem` seems unused right now.  Its former usage is commented now.  Error?  Not needed anymore?  Also this is weird.  Can't its job be done with `hoverItemIndex`?  Why this copy?